[GTK] Remove GLX support

carlosgcampos opened the pull request:
#### 320560f9e53ddcd53954059bd005e0c75eb91abf
<pre>
[GTK] Remove GLX support
<a href="https://bugs.webkit.org/show_bug.cgi?id=254016">https://bugs.webkit.org/show_bug.cgi?id=254016</a>

Reviewed by Žan Doberšek.

We are now using EGL also on X11, keeping the GLX implementation as a
fallback, that is likely unused.

* LayoutTests/platform/glib/TestExpectations:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/PlatformGTK.cmake:
* Source/WebCore/PlatformWPE.cmake:
* Source/WebCore/SourcesGTK.txt:
* Source/WebCore/platform/graphics/GLContext.cpp:
(WebCore::GLContext::createContextForWindow):
(WebCore::GLContext::createSharingContext):
* Source/WebCore/platform/graphics/GLContext.h:
* Source/WebCore/platform/graphics/PlatformDisplay.cpp:
(WebCore::PlatformDisplay::sharedDisplayDidClose):
* Source/WebCore/platform/graphics/PlatformDisplay.h:
* Source/WebCore/platform/graphics/glx/GLContextGLX.cpp: Removed.
* Source/WebCore/platform/graphics/glx/GLContextGLX.h: Removed.
* Source/WebCore/platform/graphics/gstreamer/PlatformDisplayGStreamer.cpp:
(createGstGLDisplay):
(PlatformDisplay::tryEnsureGstGLContext const):
* Source/WebCore/platform/graphics/x11/PlatformDisplayX11.cpp:
(WebCore::PlatformDisplayX11::PlatformDisplayX11):
(WebCore::PlatformDisplayX11::~PlatformDisplayX11):
(WebCore::clearSharingGLContextAtExit): Deleted.
(WebCore::PlatformDisplayX11::supportsGLX const): Deleted.
* Source/WebCore/platform/graphics/x11/PlatformDisplayX11.h:
* Source/WebCore/platform/graphics/x11/XUniquePtr.cpp: Removed.
* Source/WebCore/platform/graphics/x11/XUniquePtr.h:
* Source/WebCore/platform/graphics/x11/XUniqueResource.cpp:
(WebCore::XUniqueResource&lt;XResource::GLXPbuffer&gt;::deleteXResource): Deleted.
(WebCore::XUniqueResource&lt;XResource::GLXPixmap&gt;::deleteXResource): Deleted.
* Source/WebCore/platform/graphics/x11/XUniqueResource.h:
* Source/WebKit/UIProcess/API/glib/WebKitProtocolHandler.cpp:
(WebKit::openGLAPI):
(WebKit::WebKitProtocolHandler::handleGPU):
* Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreX11.cpp:
(WebKit::AcceleratedBackingStoreX11::checkRequirements):
(WebKit::AcceleratedBackingStoreX11::~AcceleratedBackingStoreX11):
(WebKit::AcceleratedBackingStoreX11::update):
(WebKit::glxErrorCode): Deleted.
* Source/cmake/FindOpenGL.cmake:
* Source/cmake/OptionsGTK.cmake:
* Tools/Scripts/generate-bundle:
* Tools/Scripts/webkitpy/port/xvfbdriver.py:
(XvfbDriver._xvfb_run):
* Tools/Scripts/webkitpy/port/xvfbdriver_unittest.py:
(XvfbDriverTest.test_xvfb_start_and_ready):
(XvfbDriverTest.test_xvfb_start_arbitrary_worker_number):

Canonical link: <a href="https://commits.webkit.org/261741@main">https://commits.webkit.org/261741@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/566534f4ddf11d6b44795bc3460ed3611136fdb4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [  ~~🧪 style~~](https://ews-build.webkit.org/#/builders/6/builds/112701 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/21863 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/90/builds/1378 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/4482 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/121233 "Built successfully") 
| [  ~~🧪 bindings~~](https://ews-build.webkit.org/#/builders/11/builds/116770 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/23203 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/13028 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/86/builds/5639 "2 flakes 109 failures") | 
| [  ~~🧪 webkitperl~~](https://ews-build.webkit.org/#/builders/19/builds/118470 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/17237 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/100473 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/105786 "Built successfully") | 
| [  ~~🧪 webkitpy~~](https://ews-build.webkit.org/#/builders/5/builds/116130 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/99189 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/88/builds/1017 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/46251 "14 flakes 122 failures 1 missing results") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/100994 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/14182 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/89/builds/1057 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/95461 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/83/builds/12304 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/14870 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/10403 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/102495 "Built successfully") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/20200 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/53049 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/31996 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/8191 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/16715 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/110537 "Built successfully") | 
| | | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/27298 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

zdobersek commented on Tools/Scripts/webkitpy/port/xvfbdriver.py:
```diff
@@ -93,7 +93,7 @@ def _xvfb_close_pipe(self, pipe_fds):
     def _xvfb_run(self, environment):
         read_fd, write_fd = self._xvfb_pipe()
         run_xvfb = ['Xvfb', '-displayfd', str(write_fd), '-nolisten', 'tcp',
-                    '+extension', 'GLX', '-ac', '-screen', '0',
```
Does this affect GTK? Isn't that still deciding on its own whether or not to use GLX?

carlosgcampos replied:
Yes, but we don't use OpenGL in the UI process, we use a cairo xlib surface rendered with cairo.

carlosgcampos committed:
[GTK] Remove GLX support
https://bugs.webkit.org/show_bug.cgi?id=254016

Reviewed by Žan Doberšek.

We are now using EGL also on X11, keeping the GLX implementation as a
fallback, that is likely unused.

* LayoutTests/platform/glib/TestExpectations:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/PlatformGTK.cmake:
* Source/WebCore/PlatformWPE.cmake:
* Source/WebCore/SourcesGTK.txt:
* Source/WebCore/platform/graphics/GLContext.cpp:
(WebCore::GLContext::createContextForWindow):
(WebCore::GLContext::createSharingContext):
* Source/WebCore/platform/graphics/GLContext.h:
* Source/WebCore/platform/graphics/PlatformDisplay.cpp:
(WebCore::PlatformDisplay::sharedDisplayDidClose):
* Source/WebCore/platform/graphics/PlatformDisplay.h:
* Source/WebCore/platform/graphics/glx/GLContextGLX.cpp: Removed.
* Source/WebCore/platform/graphics/glx/GLContextGLX.h: Removed.
* Source/WebCore/platform/graphics/gstreamer/PlatformDisplayGStreamer.cpp:
(createGstGLDisplay):
(PlatformDisplay::tryEnsureGstGLContext const):
* Source/WebCore/platform/graphics/x11/PlatformDisplayX11.cpp:
(WebCore::PlatformDisplayX11::PlatformDisplayX11):
(WebCore::PlatformDisplayX11::~PlatformDisplayX11):
(WebCore::clearSharingGLContextAtExit): Deleted.
(WebCore::PlatformDisplayX11::supportsGLX const): Deleted.
* Source/WebCore/platform/graphics/x11/PlatformDisplayX11.h:
* Source/WebCore/platform/graphics/x11/XUniquePtr.cpp: Removed.
* Source/WebCore/platform/graphics/x11/XUniquePtr.h:
* Source/WebCore/platform/graphics/x11/XUniqueResource.cpp:
(WebCore::XUniqueResource<XResource::GLXPbuffer>::deleteXResource): Deleted.
(WebCore::XUniqueResource<XResource::GLXPixmap>::deleteXResource): Deleted.
* Source/WebCore/platform/graphics/x11/XUniqueResource.h:
* Source/WebKit/UIProcess/API/glib/WebKitProtocolHandler.cpp:
(WebKit::openGLAPI):
(WebKit::WebKitProtocolHandler::handleGPU):
* Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreX11.cpp:
(WebKit::AcceleratedBackingStoreX11::checkRequirements):
(WebKit::AcceleratedBackingStoreX11::~AcceleratedBackingStoreX11):
(WebKit::AcceleratedBackingStoreX11::update):
(WebKit::glxErrorCode): Deleted.
* Source/cmake/FindOpenGL.cmake:
* Source/cmake/OptionsGTK.cmake:
* Tools/Scripts/generate-bundle:
* Tools/Scripts/webkitpy/port/xvfbdriver.py:
(XvfbDriver._xvfb_run):
* Tools/Scripts/webkitpy/port/xvfbdriver_unittest.py:
(XvfbDriverTest.test_xvfb_start_and_ready):
(XvfbDriverTest.test_xvfb_start_arbitrary_worker_number):

Canonical link: https://commits.webkit.org/261741@main
